src/rulesets/Base/arraymath.jl

```diff
@@ -94,6 +94,77 @@ function rrule(
     return A * B, times_pullback
 end
 
+#####
+##### fused 3-argument *
+#####
+
+using LinearAlgebra: mat_mat_scalar, mat_vec_scalar, rmul!, StridedMaybeAdjOrTransMat
+
+function rrule(
+    ::typeof(mat_mat_scalar),
+    A::StridedMaybeAdjOrTransMat{<:CommutativeMulNumber},
+    B::StridedMaybeAdjOrTransMat{<:CommutativeMulNumber},
+    γ::CommutativeMulNumber
+)
+    AB  = mat_mat_scalar(A, B, one(γ))
+    function mat_mat_scalar_back(Ȳ)
+        return (
+            NO_FIELDS,
+            InplaceableThunk(
+                @thunk(mat_mat_scalar(Ȳ, B', conj(γ))),
+                dA -> mul!(dA, Ȳ, B', conj(γ), true)
+            ),
+            InplaceableThunk(
+                @thunk(mat_mat_scalar(A', Ȳ, conj(γ))),
+                dB -> mul!(dB, A', Ȳ, conj(γ), true)
+            ),
+            @thunk if iszero(γ)
+                dot(AB, Ȳ)
+            else
+                dot(AB, Ȳ) / conj(γ)  # re-use mutated AB
+            end
+        )
+    end
+    C = if iszero(γ)
+        zero(AB)
+    else
+        rmul!(AB, γ)
+    end
+    return C, mat_mat_scalar_back
+end
+
+function rrule(
+    ::typeof(mat_vec_scalar),
+    A::StridedMaybeAdjOrTransMat{<:CommutativeMulNumber},
+    b::StridedVector{<:CommutativeMulNumber},
+    γ::CommutativeMulNumber
+)
+    Ab  = mat_vec_scalar(A, b, one(γ))
```

Comment:
```suggestion
    Ab  = A * b
```
Why do we need the `mat_vec_scalar`? And similarly in the other rule

Reply:
If I want to later mutate `rmul!(AB, γ)`, then I need to ensure its type is wide enough. Stumbled into this by accident but put a case in the tests where γ is the only complex number.